File: docs-examples/example-python/src/test/python/micronaut/configuration/openapi/docs/HelloControllerTest.py
from micronaut.test.extensions.junit5.annotation import MicronautTest
from org.junit.jupiter.api import Test

from . import OpenApiSpec


@MicronautTest
class HelloControllerTest:

    @Test
    def test_open_api_definition(self):
        open_api = OpenApiSpec.load()

        info = open_api.getInfo()
        assert info.getTitle() == "Hello World"
        assert info.getDescription() == "My API"
        assert info.getVersion() == "0.0"
        assert info.getContact().getName() == "Fred"
        assert info.getContact().getUrl() == "https://gigantic-server.com"
        assert info.getContact().getEmail() == "Fred@gigagantic-server.com"
        assert info.getLicense().getName() == "Apache 2.0"
        assert info.getLicense().getUrl() == "https://foo.bar"

    @Test
    def test_docstring_fills_the_descriptions(self):
        operation = OpenApiSpec.load().getPaths().get("/hello/{name}").getGet()

        assert operation.getOperationId().startswith("index")
        parameter = operation.getParameters().get(0)
        assert parameter.getName() == "name"
        assert parameter.getIn() == "path"
        assert parameter.getDescription() == "The person's name"
        assert parameter.getRequired()
        assert parameter.getSchema().getType() == "string"
        response = operation.getResponses().get("200")
        assert response.getDescription() == "The greeting"
        assert response.getContent().get("text/plain").getSchema().getType() == "string"

File: docs-examples/example-java/src/test/java/io/micronaut/configuration/openapi/docs/HelloControllerTest.java
package io.micronaut.configuration.openapi.docs;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class HelloControllerTest {

    @Test
    void openApiDefinition() {
        var openApi = OpenApiSpec.load();

        var info = openApi.getInfo();
        assertEquals("Hello World", info.getTitle());
        assertEquals("My API", info.getDescription());
        assertEquals("0.0", info.getVersion());
        assertEquals("Fred", info.getContact().getName());
        assertEquals("https://gigantic-server.com", info.getContact().getUrl());
        assertEquals("Fred@gigagantic-server.com", info.getContact().getEmail());
        assertEquals("Apache 2.0", info.getLicense().getName());
        assertEquals("https://foo.bar", info.getLicense().getUrl());
    }

    @Test
    void javadocFillsTheDescriptions() {
        var operation = OpenApiSpec.load().getPaths().get("/hello/{name}").getGet();

        assertTrue(operation.getOperationId().startsWith("index"));
        var parameter = operation.getParameters().get(0);
        assertEquals("name", parameter.getName());
        assertEquals("path", parameter.getIn());
        assertEquals("The person's name", parameter.getDescription());
        assertTrue(parameter.getRequired());
        assertEquals("string", parameter.getSchema().getType());
        var response = operation.getResponses().get("200");
        assertEquals("The greeting", response.getDescription());
        assertEquals("string", response.getContent().get("text/plain").getSchema().getType());
    }
}
